pkg/systemd/services/timer-dialog.jsx

```diff
@@ -268,49 +398,72 @@
+                                           helperTextInvalid={repeatPatterns[idx].helperTextInvalid}>
                                     <Flex className="specific-repeat-group" data-index={idx}>
                                         {repeat == "minutely" &&
                                             <TextInput className='delay-number'
                                                        id={repeat}
+                                                       isRequired
                                                        value={repeatPatterns[idx].second}
+                                                       onBlur={repeatValidate}
                                                        onChange={second => {
                                                            const arr = [...repeatPatterns];
                                                            arr[idx].second = second;
                                                            setRepeatPatterns(arr);
                                                        }}
-                                                       validated={submitted && validationFailedSecond ? "error" : "default"} />
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
                                         }
                                         {repeat == "hourly" && <>
                                             <TextInput className='delay-number'
+                                                       id={repeat}
+                                                       isRequired
                                                        value={repeatPatterns[idx].minute}
+                                                       onBlur={repeatValidate}
                                                        onChange={minute => {
                                                            const arr = [...repeatPatterns];
                                                            arr[idx].minute = minute;
                                                            setRepeatPatterns(arr);
                                                        }}
-                                                       validated={submitted && validationFailedMinute ? "error" : "default"} />
+                                                       pattern='^[0-9]+$'
+                                                       type='number'
+                                                       min='0'
+                                                       max='59'
+                                                       validated={repeatPatterns[idx].validated} />
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)